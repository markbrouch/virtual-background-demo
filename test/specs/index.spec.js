import { shallowMount } from '@vue/test-utils'

import IndexPage from '~/pages/index.vue'

jest.mock('~/composables/useLocalVideoTrack', () => ({
  __esModule: true,
  default: () => ({
    localVideoTrack: null,
    getLocalVideoTrack: jest.fn(),
  }),
}))

jest.mock('~/composables/useVirtualBackground', () => ({
  __esModule: true,
  default: () => ({
    startVirtualBackground: jest.fn(),
  }),
}))

describe('Index Page', () => {
  it('should render correctly', () => {
    const wrapper = shallowMount(IndexPage)

    expect(wrapper.element).toMatchSnapshot()
  })
})
