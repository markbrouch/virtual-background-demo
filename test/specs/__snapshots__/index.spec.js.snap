// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Page should render correctly 1`] = `
<div>
  <v-row-stub
    tag="div"
  >
    <v-col-stub
      tag="div"
    >
      <!---->
       
      <h1>
        Source
      </h1>
    </v-col-stub>
     
    <v-col-stub
      tag="div"
    >
      <!---->
       
      <h1>
        Final
      </h1>
    </v-col-stub>
  </v-row-stub>
   
  <v-row-stub
    tag="div"
  >
    <v-col-stub
      tag="div"
    >
      <v-btn-stub
        activeclass=""
        tag="button"
        type="button"
      >
        
        Show Composite Steps
      
      </v-btn-stub>
    </v-col-stub>
  </v-row-stub>
   
  <!---->
</div>
`;
